### PR TITLE
docs: Add some clarity to ottrace docs

### PR DESCRIPTION
**Pull request by arielvalentin**
Fixes https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/521

### CODE REVIEW

File: propagator/ottrace/README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ The `opentelemetry-propagator-ottrace` gem contains injectors and extractors for
 
 | Header Name         | Description                                                                                                                            | Required              |
 | ------------------- | -------------------------------------------------------------------------------------------------------------------------------------- | --------------------- |
-| `ot-tracer-traceid` | uint64 encoded as a string of 16 hex characters                                                                                        | yes                   |
-| `ot-tracer-spanid`  | uint64 encoded as a string of 16 hex characters                                                                                        | yes                   |
+| `ot-tracer-traceid` | 64-bit; 16 hex digits or 128-bit; 32 hex digits                                                          | yes                   |
```
Comment (mwear) [inline propagator/ottrace/README.md:10]:
```suggestion
| `ot-tracer-traceid` | 64-bit; 16 hex digits                                              | yes                   |
```

Comment (mwear) [reply propagator/ottrace/README.md:10]: `ot-tracer-traceid` will always be 64-bits on the wire. double check, but i believe this test case confirms that: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/propagator/ottrace/test/opentelemetry/propagator/ottrace/text_map_propagator_test.rb#L311-L320

**Review by mwear — CHANGES_REQUESTED** [review]

**Comment by arielvalentin** [comment]
@mwear hold on... I committed your changes a little too soon

> This propagator is compatible with 8 or 16 byte trace ids which it normalizes to 16 bytes but only emits 8 byte trace ids.

8 bytes (64 bit) or 16 bytes (128 bit) `extract` but always 8 bytes (64 bit) `inject`

**Comment by mwear** [comment]
> 8 bytes (64 bit) or 16 bytes (128 bit) extract but always 8 bytes (64 bit) inject

:point_up: is correct. I'm not sure if we should clarify the inject vs extract case, if you just want to go with what we originally had (before my suggestion).

File: propagator/ottrace/README.md [hunk 2]
```diff
@@ -7,8 +7,8 @@ The `opentelemetry-propagator-ottrace` gem contains injectors and extractors for
 
 | Header Name         | Description                                                                                                                            | Required              |
 | ------------------- | -------------------------------------------------------------------------------------------------------------------------------------- | --------------------- |
-| `ot-tracer-traceid` | uint64 encoded as a string of 16 hex characters                                                                                        | yes                   |
-| `ot-tracer-spanid`  | uint64 encoded as a string of 16 hex characters                                                                                        | yes                   |
+| `ot-tracer-traceid` | 64-bit; 16 hex digits                                              | yes                   |
```
Comment (arielvalentin) [inline propagator/ottrace/README.md:10]:
```suggestion
| `ot-tracer-traceid` | 64-bit; 16 hex digits (extract and inject) or 128-bit; 32 hex digits (extract)                                             | yes                   |
```

File: propagator/ottrace/README.md [hunk 3]
```diff
@@ -24,11 +24,15 @@ This issue was [fixed](https://github.com/open-telemetry/opentelemetry-go-contri
...
-`3c3039f4d78d5c02ee8e3e41b17ce105` would be truncated to `ee8e3e41b17ce105`.
+OTTrace was changed to be interoperable with other format so it is supposed to 8 or 16 byte array values for the trace-id.
+
+In order to do that Lightstep released a version of the OTTrace propagators in OpenTracing SDKs that left padded 8 byte headers to 16 bytes using an additional 8 bytes of 0s.
+
+The reality of the world is not every application upgraded to support 16 byte array propagation format, but this propagator must still convert legacy 8 byte trace ids to match the W3C Trace Context Trace ID 16 byte array.
+
+In addition to that it must be interoperable with legacy OTTracers, which expect 8 byte headers so this propagator truncates the value from a 16 byte array to an 8 byte array value before inject it into the carrier.
+
+This propagator is compatible with 8 or 16 byte trace ids which it normalizes to 16 bytes but only emits 8 byte trace ids.
```
Comment (arielvalentin) [inline propagator/ottrace/README.md:35]:
```suggestion
This propagator is compatible with 64-bit or 128-bit trace ids when extracting the parent context, however it truncates the trace ids down to 64-bit trace ids when injecting the context.
```

File: propagator/ottrace/README.md [hunk 4]
```diff
@@ -24,11 +24,15 @@ This issue was [fixed](https://github.com/open-telemetry/opentelemetry-go-contri
...
 ### Interop and trace ids
 
-The OT trace propagation format expects trace ids to be 64-bits. In order to
-interop with OpenTelemetry, trace ids need to be truncated to 64-bits before
-sending them on the wire. When truncating, the least significant (right-most)
-bits MUST be retained. For example, a trace id of
-`3c3039f4d78d5c02ee8e3e41b17ce105` would be truncated to `ee8e3e41b17ce105`.
+OTTrace was changed to be interoperable with other format so it is supposed to 8 or 16 byte array values for the trace-id.
+
+In order to do that Lightstep released a version of the OTTrace propagators in OpenTracing SDKs that left padded 8 byte headers to 16 bytes using an additional 8 bytes of 0s.
```
Comment (arielvalentin) [inline propagator/ottrace/README.md:29]:
```suggestion
In order to do that Lightstep released a version of the OTTrace propagators in OpenTracing SDKs that left padded 64-bit headers to 128-bits using an additional 64-bit of 0s.
```

File: propagator/ottrace/README.md (hunk 2 above)
Comment (mwear) [reply propagator/ottrace/README.md:10]: LGTM

File: propagator/ottrace/README.md [hunk 5]
```diff
@@ -24,11 +24,15 @@ This issue was [fixed](https://github.com/open-telemetry/opentelemetry-go-contri
...
+
+The reality of the world is not every application upgraded to support 16 byte array propagation format, but this propagator must still convert legacy 8 byte trace ids to match the W3C Trace Context Trace ID 16 byte array.
```
Comment (arielvalentin) [inline propagator/ottrace/README.md:31]:
```suggestion
The reality of the world is not every application upgraded to support 16 byte array propagation format, but this propagator must still convert legacy 64 but trace ids to match the W3C Trace Context Trace ID 16 byte array.
```

File: propagator/ottrace/README.md (hunk 3 above)
Comment (mwear) [reply propagator/ottrace/README.md:35]: also LGTM

File: propagator/ottrace/README.md [hunk 6]
```diff
@@ -24,11 +24,15 @@ This issue was [fixed](https://github.com/open-telemetry/opentelemetry-go-contri
...
+
+In addition to that it must be interoperable with legacy OTTracers, which expect 8 byte headers so this propagator truncates the value from a 16 byte array to an 8 byte array value before inject it into the carrier.
```
Comment (arielvalentin) [inline propagator/ottrace/README.md:33]:
```suggestion
In addition to that it must be interoperable with legacy OTTracers, which expect 64-bit headers so this propagator truncates the value from a 128-bit to a 64-bit value before inject it into the carrier.
```

File: propagator/ottrace/README.md (hunk 5 above)
Comment (mwear) [inline propagator/ottrace/README.md:31]:
```suggestion
The reality of the world is not every application upgraded to support 16 byte array propagation format, but this propagator must still convert legacy 64-bit trace ids to match the W3C Trace Context Trace ID 16 byte array.
```

**Review by mwear — APPROVED** [review]

**Pull request merged**